### PR TITLE
Backed token with auto accrual and multiplier

### PR DESCRIPTION
**Pull request by MiniRoman**

### CODE REVIEW

File: contracts/ERC20UpgradeableWithMultiplier.sol [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+             / 1e18;
+    }
+
+    /**
+     * @notice Moves `_sharesAmount` shares from `from` to `to`.
+     *
+     * Requirements:
+     *
+     * - `from` cannot be the zero address.
+     * - `from` must hold at least `_sharesAmount` shares.
```
Comment (jdrskr) [inline contracts/ERC20UpgradeableWithMultiplier.sol:141]: * - `to` cannot be the zero address.

File: contracts/ERC20UpgradeableWithMultiplier.sol [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+    function getUnderlyingAmountByShares(uint256 _sharesAmount) public view returns (uint256) {
+        return _sharesAmount
+             * _multiplier
+             / 1e18;
+    }
+
+    /**
+     * @notice Moves `_sharesAmount` shares from `from` to `to`.
+     *
+     * Requirements:
```
Comment (jdrskr) [inline contracts/ERC20UpgradeableWithMultiplier.sol:138]: let's keep comments alligned, if other functions have emits part it should be defined here as well

File: contracts/ERC20UpgradeableWithMultiplier.sol [hunk 3]
```diff
@@ -0,0 +1,261 @@
...
+     */
+    function _mintShares(address account, uint256 sharesAmount) internal virtual {
+        require(account != address(0), "ERC20: mint to the zero address");
+        uint256 amount = getUnderlyingAmountByShares(sharesAmount);
+
+        _beforeTokenTransfer(address(0), account, amount);
+
+        _totalShares += sharesAmount;
+        _shares[account] += sharesAmount;
+        emit Transfer(address(0), account, amount);
```
Comment (jdrskr) [inline contracts/ERC20UpgradeableWithMultiplier.sol:206]: there is no need for `MintShares`/`BurnShares` events?

File: contracts/BackedTokenImplementationWithMultiplierAndAutoFeeAccrual.sol [hunk 4]
```diff
@@ -0,0 +1,565 @@
...
+ *
+ * This token contract is following the ERC20 standard.
+ * It inherits ERC20PermitDelegateTransferWithMultiplier.sol, which extends the basic ERC20 to also allow permit, delegateTransfer and delegateTransferShares EIP-712 functionality.
+ * Enforces Sanctions List via the Chainalysis standard interface.
+ * The contract contains four roles:
+ *  - A minter, that can mint new tokens.
+ *  - A burner, that can burn its own tokens, or contract's tokens.
+ *  - A pauser, that can pause or restore all transfers in the contract.
+ *  - A multiplierUpdated, that can update value of a multiplier.
+ *  - An owner, that can set the four above, and also the sanctionsList pointer.
```
Comment (jdrskr) [inline contracts/BackedTokenImplementationWithMultiplierAndAutoFeeAccrual.sol:54]: `multiplierUpdater`

File: contracts/BackedTokenImplementationWithMultiplierAndAutoFeeAccrual.sol [hunk 5]
```diff
@@ -0,0 +1,565 @@
...
+     *
+     * @param newMultiplierUpdater The address of the new multiplier updater
+     */
+    function setMultiplierUpdater(address newMultiplierUpdater) external onlyOwner {
+        multiplierUpdater = newMultiplierUpdater;
+        emit NewMultiplierUpdater(newMultiplierUpdater);
+    }
+
+    /**
+     * @dev Function to change the contract multiplier, only if oldMultiplier did not change in the meantime. Allowed only for owner
```
Comment (jdrskr) [inline contracts/BackedTokenImplementationWithMultiplierAndAutoFeeAccrual.sol:416]: Comment says owner, but it seems that multiplier updater is allowed

**Review by jdrskr — APPROVED** [review]

**Review by hemulin — APPROVED** [review]
LGTM!

**Pull request merged**